client/extensions/woocommerce/woocommerce-services/state/shipping-zone-method-settings/reducer.js

```diff
@@ -1,43 +1,34 @@
-/** @format */
-
 /**
  * Internal dependencies
  */
-
-import { createReducer } from 'state/utils';
+import { withoutPersistence } from 'state/utils';
 import {
 	WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST,
 	WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST_SUCCESS,
 } from 'woocommerce/woocommerce-services/state/action-types';
 import { LOADING } from 'woocommerce/state/constants';
 import { WOOCOMMERCE_SHIPPING_ZONE_METHOD_UPDATED } from 'woocommerce/state/action-types';
 
-const reducers = {};
-
-reducers[ WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST ] = (
-	state,
-	{ instanceId }
-) => {
-	return {
-		...state,
-		[ instanceId ]: LOADING,
-	};
-};
+export default withoutPersistence( function( state = {}, action ) {
+	switch ( action.type ) {
+		case WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST:
+			return {
+				...state,
+				[ action.instanceId ]: LOADING,
+			};
 
-reducers[ WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST_SUCCESS ] = (
-	state,
-	{ instanceId }
-) => {
-	return {
-		...state,
-		[ instanceId ]: true,
-	};
-};
+		case WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST_SUCCESS:
+			return {
+				...state,
+				[ action.instanceId ]: true,
+			};
 
-reducers[ WOOCOMMERCE_SHIPPING_ZONE_METHOD_UPDATED ] = ( state, { data } ) => {
-	return reducers[ WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST_SUCCESS ]( state, {
-		instanceId: data.id,
-	} );
-};
+		case WOOCOMMERCE_SHIPPING_ZONE_METHOD_UPDATED:
+			return {
+				...state,
+				instanceId: action.data.id,
```

Comment:
I don't think this is correct. The original code is reusing the reducer for `WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST_SUCCESS`, and passing in the `instanceId` as a parameter. It's not modifying the state with `instanceId`.

I think the following code should work instead:

```suggestion
				[ action.data.id ]: true,
```

by duplicating the extremely basic functionality in `WOOCOMMERCE_SERVICES_SHIPPING_ZONE_METHOD_SETTINGS_REQUEST_SUCCESS` instead of reusing it.

Reply:
You're right 🙂 

Thanks for catching that, I've applied a fix.